src/StatsdClient/Bufferize/StatsBufferize.cs

```diff
@@ -62,15 +63,15 @@ public void OnNewValue(Stats stats)
                 using (stats)
                 {
                     _statsRouter.Route(stats);
-                    _stopwatch = null;
+                    _resetTimer = true;
                 }
             }
 
             public bool OnIdle()
             {
-                if (_stopwatch == null)
+                if (_resetTimer)
                 {
-                    _stopwatch = System.Diagnostics.Stopwatch.StartNew();
+                    _stopwatch.Restart();
```

Comment:
You must call `_resetTimer = false;`

Reply:
My bad. Fixed in https://github.com/DataDog/dogstatsd-csharp-client/pull/142/commits/0d739f406b434b50ae2f8d9d6fc1ad62699ae4e1